src/e3/encoding/vex.py

```diff
@@ -0,0 +1,902 @@
+"""Vulnerabilities Exploitability Exchange specifications.
+
+This specification package is based on
+https://www.cisa.gov/sites/default/files/2023-04/minimum-requirements-for-vex-508c.pdf.
+
+The document
+https://www.cisa.gov/sites/default/files/2023-01/VEX_Use_Cases_Aprill2022.pdf
+has also been used for a better understanding of some VEX implementations.
+"""
+
+from __future__ import annotations
+
+from datetime import datetime, timezone
+from dateutil.parser import parse as date_parse
+from enum import Enum
+from typing import TYPE_CHECKING, TypeVar
+from uuid import uuid4
+
+from e3.json import JsonData
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    DocumentSelf = TypeVar("DocumentSelf", bound="Document")
+    MetadataSelf = TypeVar("MetadataSelf", bound="Metadata")
+    ProductSelf = TypeVar("ProductSelf", bound="Product")
+    StatementSelf = TypeVar("StatementSelf", bound="Statement")
+    StatementStatusSelf = TypeVar("StatementStatusSelf", bound="StatementStatus")
+    SubProductIdSelf = TypeVar("SubProductIdSelf", bound="SubProductId")
+
+
+class ActionOrImpact(JsonData):
+    """An object to represent an action or impact for a statement status.
+
+    For *status* :attr:`Status.NOT_AFFECTED`, if *justification* is not
+    provided, then a VEX statement **MUST** provide an *impact_statement* that
+    further explains how or why the listed *product_id* s are *not affected* by
+    *vul_id*.
+
+    If *justification* is provided, then a VEX statement **MAY** provide an
+    *impact_statement*.
+
+    :ivar str | None statement: The statement of this action or impact.
+    :ivar datetime | str | None timestamp: The time at which the statement has
+        been last modified.
+    """  # noqa RST304
+
+    def __init__(
+        self,
+        statement: str | None = None,
+        timestamp: datetime | str | None = None,
+    ):
+        self.timestamp: datetime | None
+        self.statement: str | None = statement
+        if isinstance(timestamp, str):
+            self.timestamp = date_parse(timestamp)
+        elif self.statement and not timestamp:
+            self.timestamp = datetime.now(timezone.utc).replace(microsecond=0)
+        elif isinstance(timestamp, datetime):
+            self.timestamp = timestamp.replace(microsecond=0)
+        else:
+            # Should be None
+            self.timestamp = None
+
+    def __bool__(self) -> bool:
+        """Check if this action or impact is defined.
+
+        An action or impact without a statement is considered as False.
+
+        :return: A :class:`python:bool` set to **True** if this action or impact
```

Comment:
Description is a bit too convoluated. Just say: return True if ... else return False. It's clear for everyone